.github/workflows/kb_sdk-actions.yml

```diff
@@ -0,0 +1,136 @@
+name: Basic kb-sdk tests
+
+on:
+  [push, pull_request]
+
+jobs:
+
+  test_kb-sdk_builds:
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        language: [perl, python, java]
+        test_type: [base, example]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Build with Ant
+        run: |
+          make
+
+      - name: Add bin to $PATH
+        run: |
+          echo "::add-path::$GITHUB_WORKSPACE/bin"
+
+      - name: checking kb-sdk functions
+        env:
+          KBASE_TEST_TOKEN: ${{ secrets.KBASE_TEST_TOKEN }}
+          LANGUAGE_TOKEN: ${{ matrix.language }}
+        run: |
+          env
+          kb-sdk help
+          kb-sdk version
+
+      - name: init base repo
+        if: matrix.test_type == 'base'
+        env:
+          LANGUAGE_TOKEN: ${{ matrix.language }}
+        run: |
+          kb-sdk init -l $LANGUAGE_TOKEN -u user SampleApp
+
+      - name: init example repo
+        if: matrix.test_type == 'example'
+        env:
+          LANGUAGE_TOKEN: ${{ matrix.language }}
+        run: |
+          kb-sdk init -l $LANGUAGE_TOKEN -u user --example SampleApp
+
+      - name: test ${{ matrix.test_type }} repo
+        env:
+          KBASE_TEST_TOKEN: ${{ secrets.KBASE_TEST_TOKEN }}
+          LANGUAGE_TOKEN: ${{ matrix.language }}
+        run: |
+          cd SampleApp
+          kb-sdk test || true
+          sed -i "s/test_token=/test_token=$KBASE_TEST_TOKEN/" test_local/test.cfg
+          kb-sdk test
+
+      - name: make resulting app available as artefact in case of failure
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: kbaseapp-${{ matrix.language }}-${{ matrix.test_type }}-${{ matrix.os }}
```

Comment:
I'm guessing each build overwrites the previous build here. Would it make sense to put a timestamp in the name so that doesn't happen, assuming I'm right?

Reply:
This matrix has variables `language`, `test_type`, and `os`, so the name suggested will be unique.

Reply:
Only per build, not over time, which is my point. Unless I'm not understanding what you're saying.

Reply:
Each workflow run has a separate set of artefacts - you download them from the workflow run page. E.g. here's a workflow with failed tests: https://github.com/ialarmedalien/kb_sdk/actions/runs/137590374

Reply:
To clarify, I'm talking about the uploads from PRn+1 overwriting the uploads from PRn.

Reply:
Every run of the GitHub Actions generates a fresh set of results and artefacts. For example, all the times that I've run GitHub actions on my branches of `kb_sdk`:

https://github.com/ialarmedalien/kb_sdk/actions